From e7ddbadd1db9aefa29ee15a436494d170e475778 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Thu, 4 Sep 2025 01:13:08 +0200
Subject: [PATCH 089/133] std.elf: document which EM values are not assigned in
 gABI
X-Developer-Signature: v=1; a=openpgp-sha256; l=2464; i=xtex@astrafall.org;
 h=from:subject; bh=HLKcnzcqaW9/6np1obLLg5kWrIKrkez4p/C8nUUoqZE=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78jhkgll+HnX7e7p+q5J6cWtxyYupKv/PX7b6lJD6+
 vkqMY7cjoksDGJcDJZiiixFhg3erDrp/KLLymVh5rAygQyRFmlgYGBgYGHgy03MKzXSMdIz1TbU
 MzTUMdIxYuDiFICpNvvA8L/m0Q72XVsOqTbxSCmEuBf8Mk2aL5pvr+EaotShVb8xz5rhv//x2tO
 fpmrE+jceidHom9Z22LJY9eGG2UJWiqKbmCV+cgMA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 1ca446abc3570b5946c7efb46f4d3e4ef016c1ba)
---
 lib/std/elf.zig | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/lib/std/elf.zig b/lib/std/elf.zig
index bd9a6b52ac75..24545171eeaa 100644
--- a/lib/std/elf.zig
+++ b/lib/std/elf.zig
@@ -1165,11 +1165,11 @@ pub const EM = enum(u16) {
     S370 = 9,
     /// MIPS R3000 (and R4000) little-endian, Oct 4 1993 Draft (deprecated)
     MIPS_RS3_LE = 10,
-    /// Old version of Sparc v9, from before the ABI (deprecated)
+    /// Old version of Sparc v9, from before the ABI (not gABI assigned)
     OLD_SPARCV9 = 11,
     /// HPPA
     PARISC = 15,
-    /// Fujitsu VPP500 (also old version of PowerPC; deprecated)
+    /// Fujitsu VPP500 (also old version of PowerPC, which was not gABI assigned)
     VPP500 = 17,
     /// Sun's "v8plus"
     SPARC32PLUS = 18,
@@ -1189,7 +1189,7 @@ pub const EM = enum(u16) {
     FR20 = 37,
     /// TRW RH32
     RH32 = 38,
-    /// Motorola M*Core, aka RCE (also Fujitsu MMA)
+    /// Motorola M*Core, aka RCE (also old Fujitsu MMA, which was not gABI assigned)
     MCORE = 39,
     /// ARM
     ARM = 40,
@@ -1301,7 +1301,7 @@ pub const EM = enum(u16) {
     ARC_COMPACT = 93,
     /// Tensilica Xtensa Architecture
     XTENSA = 94,
-    /// Alphamosaic VideoCore processor (also old Sunplus S+core7 backend magic number)
+    /// Alphamosaic VideoCore processor (also old Sunplus S+core7 backend magic number, which was not gABI assigned)
     VIDEOCORE = 95,
     /// Thompson Multimedia General Purpose Processor
     TMM_GPP = 96,
@@ -1309,7 +1309,7 @@ pub const EM = enum(u16) {
     NS32K = 97,
     /// Tenor Network TPC processor
     TPC = 98,
-    /// Trebia SNP 1000 processor (also old value for picoJava; deprecated)
+    /// Trebia SNP 1000 processor (also old value for picoJava, which was not gABI assigned)
     SNP1K = 99,
     /// STMicroelectronics ST200 microcontroller
     ST200 = 100,
@@ -1341,7 +1341,7 @@ pub const EM = enum(u16) {
     ALTERA_NIOS2 = 113,
     /// National Semiconductor CRX
     CRX = 114,
-    /// Motorola XGATE embedded processor (also old value for National Semiconductor CompactRISC; deprecated)
+    /// Motorola XGATE embedded processor (also old value for National Semiconductor CompactRISC, which was not gABI assigned)
     XGATE = 115,
     /// Infineon C16x/XC16x processor
     C166 = 116,
-- 
2.52.0

